src/shared/clients/codecatalystClient.ts

```diff
@@ -591,118 +592,145 @@ class CodeCatalystClientInternal {
-                            doLog('debug', 'devenv not started (ConflictException), waiting')
-                        } else {
-                            throw new ToolkitError('Dev Environment failed to start', {
-                                code: 'BadDevEnvState',
-                                cause: err,
-                            })
-                        }
+                        // - ValidationException: "… creation has failed, cannot start"
+                        // - ConflictException: "Cannot start … because update process is still going on"
+                        //   (can happen after "Update Dev Environment")
+                        doLog('info', `devenv not started (${err.code}), waiting`)
+                        // Continue retrying...
                     }
                 } else if (resp.status === 'STOPPING') {
                     progress.report({
-                        message: localize('AWS.CodeCatalyst.devenv.resuming', 'Waiting for Dev Environment to stop...'),
+                        message: localize('AWS.CodeCatalyst.devenv.stopping', 'Waiting for Dev Environment to stop...'),
                     })
-                } else if (resp.status === 'FAILED') {
-                    throw new ToolkitError('Dev Environment failed to start', { code: 'FailedDevEnv' })
                 } else {
                     progress.report({
                         message: localize('AWS.CodeCatalyst.devenv.starting', 'Opening Dev Environment...'),
                     })
                 }
 
-                if (lastStatus[lastStatus.length - 1].status !== resp.status) {
-                    lastStatus.push({ status: resp.status, start: Date.now() })
+                const lastStatus = statuses[statuses.length - 1]?.status
+                if (lastStatus !== resp.status) {
+                    statuses.push({ status: resp.status, start: Date.now() })
+                    if (resp.status !== 'RUNNING') {
+                        doLog('debug', `devenv not started, waiting`)
+                    }
                 }
                 return resp.status === 'RUNNING' ? resp : undefined
             },
             // note: the `waitUntil` will resolve prior to the real timeout if it is refreshed
             { interval: 1000, timeout: timeout.remainingTime, truthy: true }
         )
 
-        const devenv = await waitTimeout(pollDevEnv, timeout)
+        const devenv = await waitTimeout(pollDevEnv, timeout).catch(e => {
+            const err = e as Error
+            const starts = statuses.filter(o => o.status === 'STARTING').length
+            const fails = statuses.filter(o => o.status === 'FAILED').length
+
+            if (!isUserCancelledError(e)) {
+                doLog('error', 'devenv failed to start')
+            } else {
+                doLog('info', 'devenv failed to start (user cancelled)')
+                err.message = failedStartMsg()
+                throw err
```

Comment:
Try this 
```ts
throw new ToolkitError(failedStartMsg(), { cause: err })
```

It'll still be treated as a cancel because the root error was a cancel.

Reply:
Yeah, but I was thinking that having a chain of "redundant" cancels could be unhelpful?